src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,339 @@
+# ODE_MS Fixed-step, fixed-order multi-step numerical method
+# with Adams-Bashforth-Moulton coefficients
+function ode_ab_adaptive(F, y0, tspan, order::Integer=4;adaptive_order = true,
+                                                      max_order = 13,
+                                                      reltol = 1.0e-5,
+                                                      abstol = 1.0e-8,
+                                                      minstep=abs(tspan[1] - tspan[end])/1e12,
+                                                      maxstep=abs(tspan[end] - tspan[1])/2.5,
+                                                      initstep=0.,
+                                                      facmax = 1.5,
+                                                      facmin = .1,
+                                                      fac = .8,
+                                                      norm=Base.norm,
+                                                      points =:all)
+
+    # initialization of time span, which will have variable length
+    dof = length(y0)
+    tstart = tspan[1]
+    tfinal = tspan[end]
+    t = [tstart]
+
+    # initialzation of y[n], which will have variable length as well
+    y = typeof(y0)[]
+    push!(y,y0)
+
+    # initialzation of dy[n], which will have variable length as well
+    dy = typeof(F(t[1], y[1]))[]
+    push!(y,F(t[1], y[1]))
+
+    #initialization of step size array, which will have variable length as well
+    h = initstep
+    next_h = 0.0
+    if h == 0.
+        # initial guess at a step size
+        h, tdir, dy0 = hinit(F, y0, tstart, tfinal, 3, reltol, abstol)
+    else
+        tdir = sign(tfinal - tstart)
+    end
+
+    #uncomment line below to test with fixed step size
+    #@show h = diff(tspan)
+
+    #initialization of arrays used in calculation of further steps
+    #all initialized to a order x order array, which will need to expand to
+    # n x order over course of algorithm, as we use n steps, where n is
+    #unknown
+    Tdy=typeof(F(t[1], y[1]))
+    c,g,b,Φstar, Φ = intialize_coef_arrays(max_order+3,Tdy)
+
+    #Array which holds norms of LE(n+1), the local error or truncation error, which will
+    #will used in step size selection. They correspond to norm of LE(n+1)_{k-1},
+    # LE(n+1)_{k}, E(n+1)_{k+1}, and LE(n+1)_{k+2} from Harier pgs 422-424
+    err_km1, err_k, err_kp1, err_kp2 = 0.0, 0.0, 0.0, 0.0
+
+    ##Note that due to Julia starting indexing at 1, k = steporder + 1 in what follows
+    n = 1     #`n` counts the time steps. We start at n=1, the first step
+    steporder=1 #`steporder` is the order of the AB method at the current step
+
+    timeout_const = 0 # reset for how many steps before a new step size can be taken
+    timeout = timeout_const # counter for this over the following loop
+
+    wait=0 #`wait` is used to control the order adaptivity, because the order
+           # can be increased at most every other step.
```

Comment:
I now have `timeout_VS` and `timeout_VO` where VS and VO stand for variable step and variable order, respectively.